devel/block-borders.lua

```diff
@@ -24,17 +23,14 @@ local DRAW_CHARS = {
     se = string.char(218),
     sw = string.char(191),
 }
-local VALID_SIDEBAR_MODES = utils.invert{
-    df.ui_sidebar_mode.Default,
-    df.ui_sidebar_mode.LookAround,
-}
 -- persist across script runs
 color = color or COLOR_LIGHTCYAN
 
 BlockBordersOverlay = defclass(BlockBordersOverlay, guidm.MenuOverlay)
 BlockBordersOverlay.ATTRS{
     block_size = 16,
     draw_borders = true,
+    sidebar_mode=df.ui_sidebar_mode.LookAround,
```

Comment:
There was a subtle behavioral change introduced here: previously, the script could be invoked from either the Default or LookAround modes, and would preserve the mode that was selected before. Now it always enters LookAround (k), and exits back to Default when it's done.

I don't think it really matters, but again, I am curious - is there an existing way to change sidebar modes only if the pre-selected mode isn't one of _multiple_ valid options?

Reply:
I'll look into this. The intended behavior is to restore the previous mode if it is supported by `enterSidebarMode()`. It should only be "restoring" Default if the topmost screen is a DFHack window or if the starting mode is unsupported

Reply:
The part I was specifically focusing on was the "always enters LookAround" behavior, not the "always exits back to Default". The latter is actually not correct. Fixed.